### PR TITLE
Add sourcemaps flag for github paths to commonJsSettings

### PR DESCRIPTION
**Pull request by zetashift**
Fixes https://github.com/PurpleKingdomGames/tyrian/issues/155
Shamelessly copy pasted from my smithy4s PR: `https://github.com/disneystreaming/smithy4s/pull/706`

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -35,7 +35,21 @@ lazy val commonSettings: Seq[sbt.Def.Setting[_]] = Seq(
 )
 
 lazy val commonJsSettings: Seq[sbt.Def.Setting[_]] = Seq(
-  Test / scalaJSLinkerConfig ~= { _.withModuleKind(ModuleKind.CommonJSModule) }
+  Test / scalaJSLinkerConfig ~= { _.withModuleKind(ModuleKind.CommonJSModule) },
+  scalacOptions ++= {
+    // Map the sourcemaps to github paths instead of local directories
+    val flag =
+      if (scalaVersion.value.startsWith("3")) "-scalajs-mapSourceURI"
```
Comment (davesmith00000) [inline build.sbt:42]: Copy+Pasta: Tyrian isn't released for Scala 2. :smile:

File: build.sbt [hunk 2]
```diff
@@ -35,7 +35,21 @@ lazy val commonSettings: Seq[sbt.Def.Setting[_]] = Seq(
...
+      else "-P:scalajs:mapSourceURI"
+    val localSourcesPath = baseDirectory.value.toURI
+    val headCommit       = git.gitHeadCommit.value.get
+    scmInfo.value.map { info =>
+      val remoteSourcesPath =
+        s"${info.browseUrl.toString
+            .replace("github.com", "raw.githubusercontent.com")}/$headCommit"
+      s"${flag}:$localSourcesPath->$remoteSourcesPath"
```
Comment (davesmith00000) [inline build.sbt:50]: Gosh. I would never have thought of that. Thanks!

**Review by davesmith00000 — DISMISSED** [review]

File: build.sbt (hunk 1 above)
Comment (zetashift) [reply build.sbt]: Another reason why I should never copy paste without due diligence :P

File: build.sbt (hunk 2 above)
Comment (zetashift) [reply build.sbt]: Same here, all credit to the excellent [sbt-typelevel](https://github.com/typelevel/sbt-typelevel) plugin

**Review by davesmith00000 — APPROVED** [review]
LGTM! Thanks. :smile:

**Pull request merged**